physics/ugwpv1_gsldrag.F90

```diff
@@ -0,0 +1,720 @@
+       print *,  ' ccpp do_gsl_drag_tofd  flag ',     do_gsl_drag_tofd
+
+       print *, ' ccpp: ugwpv1_gsldrag_init  '
+    endif
+
+
+
+    is_initialized = .true.
+
+
+    end subroutine ugwpv1_gsldrag_init
+
+
+! -----------------------------------------------------------------------
+! finalize of ugwpv1_gsldrag   (_finalize)
+! -----------------------------------------------------------------------
+
+!>@brief The subroutine finalizes the CIRES UGWP
+
+!> \section arg_table_ugwpv1_gsldrag_finalize Argument Table
+!! \htmlinclude ugwpv1_gsldrag_finalize.html
+!!
+
+    subroutine ugwpv1_gsldrag_finalize(errmsg, errflg)
+
+    implicit none
+!
+    character(len=*), intent(out) :: errmsg
+    integer,          intent(out) :: errflg
+
+! Initialize CCPP error handling variables
+    errmsg = ''
+    errflg = 0
+
+    if (.not.is_initialized) return
+
+    call cires_ugwp_dealloc
+
+    is_initialized = .false.
+
+    end subroutine ugwpv1_gsldrag_finalize
+
+
+! -----------------------------------------------------------------------
+!    originally from ugwp_driver_v0.f
+!    driver of cires_ugwp   (_driver)
+! -----------------------------------------------------------------------
+!   driver is called after pbl & before chem-parameterizations
+! -----------------------------------------------------------------------
+!  order = dry-adj=>conv=mp-aero=>radiation -sfc/land- chem -> vertdiff-> [rf-gws]=> ion-re
+! -----------------------------------------------------------------------
+!>@brief These subroutines and modules execute the CIRES UGWP Version 0
+!>\defgroup ugwpv1_gsldrag_run Unified Gravity Wave Physics General Algorithm
+!> @{
+!! The physics of NGWs in the UGWP framework (Yudin et al. 2018 \cite yudin_et_al_2018) is represented by four GW-solvers, which is introduced in Lindzen (1981) \cite lindzen_1981, Hines (1997) \cite hines_1997, Alexander and Dunkerton (1999) \cite alexander_and_dunkerton_1999, and Scinocca (2003) \cite scinocca_2003. The major modification of these GW solvers is represented by the addition of the background dissipation of temperature and winds to the saturation criteria for wave breaking. This feature is important in the mesosphere and thermosphere for WAM applications and it considers appropriate scale-dependent dissipation of waves near the model top lid providing the momentum and energy conservation in the vertical column physics (Shaw and Shepherd 2009 \cite shaw_and_shepherd_2009). In the UGWP-v0, the modification of Scinocca (2003) \cite scinocca_2003 scheme for NGWs with non-hydrostatic and rotational effects for GW propagations and background dissipation is represented by the subroutine \ref fv3_ugwp_solv2_v0. In the next release of UGWP, additional GW-solvers will be implemented along with physics-based triggering of waves and stochastic approaches for selection of GW modes characterized by horizontal phase velocities, azimuthal directions and magnitude of the vertical momentum flux (VMF).
```

Comment:
Violates Fortran standard's line length limit of 132 characters. This will abort on some compilers (even for comments).